server/app.py

```diff
@@ -127,7 +127,7 @@ def build_stats_response(games):
                 # sum all stats for the player
                 if player.name in stats:
                     existing_data = stats[player.name]
-                    stats_to_sum = data.keys() - ['pay', 'salary', 'salary_per_point']
+                    stats_to_sum = data.keys() - ['pay', 'salary', 'salary_per_point', 'o_efficiency', 'd_efficiency', 'total_efficiency']
```

Comment:
This is the right choice because adding the efficiencies doesn't make sense but it also exposes something missing / weird from my design. This function is called to build responses for both a certain week and all time. However a few fields are inconsistent:

Week response:
* stats for the week. `stats_to_sum` doesn't really come into play since there should be only 1 line per player
* pay which is calculated from the week's stats
* salary which comes from the player model and takes into account all known data about the player (past and future)
* efficiency scores based on the current week

All Time response:
* all stats from all games summed up
* pay from the last week (not confirmed but I think this is how it will work out since the games are ordered before the response is built)
* salary which is based on all known data for the player (same as above)
* efficiency for the most recent week (related to game ordering)

I wanted to point this out because I expect we'll need to address this and break this function up as we add more things.

Reply:
Yes, I was looking along those lines which was my motivation for issue #137 . I will do a little more investigation as something is a little wonky here.

Reply:
Love this feature! 

>efficiency for the most recent week (related to game ordering)
It looks like the current values for efficiency are coming from week 1 for the stats endpoint. Players with no week 1 data have their week 2 efficiencies returned.

Reply:
For week (1..n):
- stats for that game (goals, assists, ...)
- efficiency - numbers for that game
- pay based on stats for that game (ie: a linear combination of the above)
- salary per point - for the game
- salary - cumulative salary up to that week

For All Time:
- sum(stats over all weeks)
- efficiency - sum(points_for over all weeks) / sum(points played over all weeks)
- sum(pay over all weeks)
- salary per point???
    - average over all games of salary per point
    - or -
    - sum(pay over all weeks) / sum(points over all weeks)
- salary - cumulative salary up to latest week

For Salary - which makes more sense?
- This method puts equal weight between games (eg: a 30 point game vs a 20 point game have equal weighting)
        sum(salary per point over all games) / number of games
- This method is an overall average
        sum(pay over all games) / sum(points played over all games)

Then multiply by 15

Reply:
I thought equal weight per game was interesting but I have nothing to backup why

Reply:
Having equal weight per game does minimize the effect of an extraordinary good (or bad) game.

Reply:
I think that was it. Average per game dampens things a bit. I prefer the average by game approach